### PR TITLE
Add val and var support to ZiplineService

### PR DESCRIPTION
**Pull request by adrw**
Paired with @swankjesse on adding val and support to ZiplineService Kotlin compiler code generation.

### CODE REVIEW

File: zipline-kotlin-plugin/src/main/kotlin/app/cash/zipline/kotlin/AdapterGenerator.kt [hunk 1]
```diff
@@ -806,6 +840,11 @@ internal class AdapterGenerator(
       )
     }
 
+    // if this function is part of a property, box it
```
Comment (adrw) [inline zipline-kotlin-plugin/src/main/kotlin/app/cash/zipline/kotlin/AdapterGenerator.kt]: @swankjesse do you remember if this is necessary?

Comment (swankjesse) [reply zipline-kotlin-plugin/src/main/kotlin/app/cash/zipline/kotlin/AdapterGenerator.kt]: Already done. You can remove this comment.

File: zipline-kotlin-plugin/src/main/kotlin/app/cash/zipline/kotlin/ZiplineIrGenerationExtension.kt [hunk 2]
```diff
@@ -47,6 +47,8 @@ class ZiplineIrGenerationExtension(
               currentScope!!,
               declaration
             ).generateAdapterIfAbsent()
+            // When iterating on Kotlin compiler plugin, set breakpoint here and
```
Comment (swankjesse) [inline zipline-kotlin-plugin/src/main/kotlin/app/cash/zipline/kotlin/ZiplineIrGenerationExtension.kt:50]: 👍🏻

File: zipline-kotlin-plugin/src/main/kotlin/app/cash/zipline/kotlin/ir.kt [hunk 3]
```diff
@@ -75,12 +75,31 @@ import org.jetbrains.kotlin.ir.util.defaultType
 import org.jetbrains.kotlin.name.FqName
 import org.jetbrains.kotlin.name.Name
 
+// Should produce a string like "var count: Int" or "val count: Int" or "fun echo(request: EchoRequest): EchoResponse"
```
Comment (swankjesse) [inline zipline-kotlin-plugin/src/main/kotlin/app/cash/zipline/kotlin/ir.kt]: nit: `var count: kotlin.Int` ?

File: zipline-kotlin-plugin/tests/src/test/kotlin/app/cash/zipline/kotlin/ZiplineKotlinPluginTest.kt [hunk 4]
```diff
@@ -28,6 +28,7 @@ import kotlinx.serialization.KSerializer
 import org.jetbrains.kotlin.compiler.plugin.ComponentRegistrar
 import org.junit.Test
 
+// Good for early on break points in compiler, onerous to exercise
```
Comment (swankjesse) [inline zipline-kotlin-plugin/tests/src/test/kotlin/app/cash/zipline/kotlin/ZiplineKotlinPluginTest.kt]: this note no longer fits . . . remove or expand?

File: zipline/src/engineTest/kotlin/app/cash/zipline/EndpointPropertiesTest.kt [hunk 5]
```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
```
Comment (swankjesse) [inline zipline/src/engineTest/kotlin/app/cash/zipline/EndpointPropertiesTest.kt]: 2022. (my bad)

File: zipline/src/engineTest/kotlin/app/cash/zipline/SampleValVarService.kt [hunk 6]
```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
```
Comment (swankjesse) [inline zipline/src/engineTest/kotlin/app/cash/zipline/SampleValVarService.kt]: 2022

File: zipline/src/engineTest/kotlin/app/cash/zipline/SampleValVarService.kt [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+@Serializable
+data class SampleValVarResponse(
+  val message: String
+)
+
+/**
+ * Note that this interface is unused. It only serves as a sample for what the code generators
+ * should consume and produce.
+ */
+interface SampleValVarService<T> : ZiplineService {
```
Comment (swankjesse) [inline zipline/src/engineTest/kotlin/app/cash/zipline/SampleValVarService.kt]: I think you can remove this entire file

File: zipline/testing/src/commonMain/kotlin/app/cash/zipline/testing/valsAndVars.kt [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline.testing
+
+import app.cash.zipline.ZiplineService
+
+interface ValVarService : ZiplineService {
```
Comment (swankjesse) [inline zipline/testing/src/commonMain/kotlin/app/cash/zipline/testing/valsAndVars.kt]: Unused?

**Review by swankjesse — APPROVED** [review]
LOVE IT

**Pull request merged**